eng/common/TestResources/New-TestResources.ps1

```diff
@@ -758,7 +758,8 @@ try {
     if ($TestApplicationSecret -and $ServicePrincipalAuth) {
         $templateParameters.Add('testApplicationSecret', $TestApplicationSecret)
     }
-    if ($CI -and $Environment -eq 'AzureCloud') {
+    # Only add subnets when running in an azure pipeline context
+    if ($env:SYSTEM_TEAMPROJECTID -and $Environment -eq 'AzureCloud') {
```

Comment:
Isn't that what the CI parameter is about?

Reply:
The challenge with using CI is some other things are coupled to it (parameter sets, other conditionals). Once in a while I need to actually test the CI parameter locally (e.g. if I'm making CI based behavior changes). So I prefer to be explicit here about the conditional, which allows me to still run CI mode in any environment, and restrict behavior that only works in the pipeline to that environment.

Reply:
I can change it to CI and just modify conditionals if needed when testing. I agree there are trade-offs either way.